File: fs/src/main/scala/farjs/fs/popups/DriveController.scala
package farjs.fs.popups

import scommons.react._

object DriveController extends FunctionComponent[DriveControllerProps] {

  private[popups] var drivePopup: ReactClass = DrivePopup

  protected def render(compProps: Props): ReactElement = {
    val props = compProps.plain

    props.showDrivePopupOnLeft.toOption match {
      case Some(showOnLeft) =>
        <(drivePopup)(^.plain := DrivePopupProps(
          dispatch = props.dispatch,
          onChangeDir = { dir =>
            props.onClose()
            props.onChangeDir(dir, showOnLeft)
          },
          onClose = props.onClose,
          showOnLeft = showOnLeft
        ))()
      case None => null
    }
  }
}

File: fs/popups/DriveController.mjs
/**
 * @import { Dispatch } from "@farjs/filelist/FileListData.mjs"
 */
import React from "react";
import DrivePopup from "./DrivePopup.mjs";

const h = React.createElement;

/**
 * @typedef {{
 *  readonly dispatch: Dispatch;
 *  readonly showDrivePopupOnLeft?: boolean;
 *  onChangeDir(dir: string, showOnLeft: boolean): void;
 *  onClose(): void;
 * }} DriveControllerProps
 */

/**
 * @param {DriveControllerProps} props
 */
const DriveController = (props) => {
  const { drivePopup } = DriveController;

  const showOnLeft = props.showDrivePopupOnLeft;

  return showOnLeft !== undefined
    ? h(drivePopup, {
        dispatch: props.dispatch,
        onChangeDir: (dir) => {
          props.onClose();
          props.onChangeDir(dir, showOnLeft);
        },
        onClose: props.onClose,
        showOnLeft: showOnLeft,
      })
    : null;
};

DriveController.displayName = "DriveController";
DriveController.drivePopup = DrivePopup;

export default DriveController;
